pkg/controller/pxc/users.go

```diff
@@ -326,3 +327,41 @@ func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool
 func sha256Hash(data []byte) string {
 	return fmt.Sprintf("%x", sha256.Sum256(data))
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileSyncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) {
+	if cr.Status.Status != api.AppStateReady {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	stopCh := make(chan error, 1)
+
+	go func() {
+		r.syncUsersChan <- 1
+		defer func() {
+			if len(r.syncUsersChan) > 0 {
+				<-r.syncUsersChan
+				log.Info("done")
+			}
+		}()
+
+		stopCh <- r.syncPXCUsersWithProxySQL(cr)
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			cancel()
+			if len(r.syncUsersChan) > 0 {
+				<-r.syncUsersChan
+			}
```

Comment:
Maybe can be moved to defer
Same as **https://github.com/percona/percona-xtradb-cluster-operator/pull/433/files#diff-1bd1322bf02270b23ebc29a1b08fe90dR344-R346**

Reply:
I think we don't need this because go func() have it own defer

Reply:
Yes, but we can avoid code duplication then